website/addons/dataverse/client.py

```diff
@@ -1,103 +1,121 @@
 import httplib as http
+
 from dataverse import Connection
+from dataverse.exceptions import ConnectionError, UnauthorizedError, OperationFailedError
 
 from framework.exceptions import HTTPError
 from website.addons.dataverse import settings
 
 
-def connect(username, password, host=settings.HOST):
-    connection = Connection(
-        username=username,
-        password=password,
-        host=host,
-        disable_ssl=not settings.VERIFY_SSL,
-    )
-    return connection if connection.connected else None
+def _connect(token, host=settings.HOST):
+    try:
+        return Connection(host, token)
+    except ConnectionError:
+        return None
 
 
 def connect_from_settings(user_settings):
-    return connect(
-        user_settings.dataverse_username,
-        user_settings.dataverse_password
-    ) if user_settings else None
+    try:
+        return _connect(user_settings.api_token) if user_settings else None
+    except UnauthorizedError:
+        return None
 
 
-def connect_or_403(username, password, host=settings.HOST):
-    connection = Connection(
-        username=username,
-        password=password,
-        host=host,
-        disable_ssl=not settings.VERIFY_SSL,
-    )
-    if connection.status == http.FORBIDDEN:
-        raise HTTPError(http.FORBIDDEN)
-    return connection if connection.connected else None
+def connect_or_401(token):
+    try:
+        return _connect(token)
+    except UnauthorizedError:
+        raise HTTPError(http.UNAUTHORIZED)
 
 
-def connect_from_settings_or_403(user_settings):
-    return connect_or_403(
-        user_settings.dataverse_username,
-        user_settings.dataverse_password
-    ) if user_settings else None
+def connect_from_settings_or_401(user_settings):
+    return connect_or_401(user_settings.api_token) if user_settings else None
 
 
 def delete_file(file):
```

Comment:
Removed.

Note that `get_files` is still being used in one place to determine whether the root directory should have a state selector.